### PR TITLE
feat: cache theme compile and make async theme compile more reliable

### PR DESCRIPTION
**Pull request by pweyck**

### CODE REVIEW

File: tests/ThemeTest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { test, expect } from '../src/index';
+
+test('theme compilation', async ({
```
Comment (Phil23) [inline tests/ThemeTest.spec.ts]:
```suggestion
test('Theme compilation.', async ({
```

File: src/services/ShopInfo.ts [hunk 2]
```diff
@@ -1,6 +1,27 @@
+import { Page, Request } from 'playwright-core';
 import { type AdminApiContext } from './AdminApiContext';
 
 export const isSaaSInstance = async (adminApiContext: AdminApiContext): Promise<boolean> => {
     const instanceFeatures = await adminApiContext.get('./instance/features');
     return instanceFeatures.ok();
+};
+
+export const themeIsCompiled = async (page: Page): Promise<boolean> => {
```
Comment (Phil23) [inline src/services/ShopInfo.ts]: For consistency:

```suggestion
export const isThemeCompiled = async (page: Page): Promise<boolean> => {
```

File: src/fixtures/DefaultSalesChannel.ts [hunk 3]
```diff
@@ -312,35 +276,23 @@ export const test = base.extend<NonNullable<unknown>, FixtureTypes>({
...
-                    if (json.notifications.find(n => n.message.includes(`Compilation for sales channel ${DefaultSalesChannel.salesChannel.name} completed`))) {
-                        tmpPage.close();
-                        tmpContext.close();
+                await AdminApiContext.post(
+                    `./_action/theme/${SalesChannelBaseConfig.defaultThemeId}/assign/${uuid}`
+                );
 
-                        break;
+                if (isSaasInstance) {
+                    while (!await themeIsCompiled(tmpPage)) {
```
Comment (pweyck) [inline src/fixtures/DefaultSalesChannel.ts]:
```suggestion
                    while (!await themeIsCompiled(tmpPage)) {
```
```suggestion
                    while (!await isThemeCompiled(tmpPage)) {
```

File: src/fixtures/DefaultSalesChannel.ts [hunk 4]
```diff
@@ -312,35 +276,23 @@ export const test = base.extend<NonNullable<unknown>, FixtureTypes>({
...
-                    const json = await response.json() as { notifications: [{ message: string }], timestamp: string };
+            const tmpContext = await browser.newContext({
+                baseURL: DefaultSalesChannel.url,
+            });
+            const tmpPage = await tmpContext.newPage();
 
-                    if (json.timestamp) {
-                        latestTimestamp = json.timestamp;
-                    }
+            if (!await themeIsCompiled(tmpPage)) {
```
Comment (pweyck) [inline src/fixtures/DefaultSalesChannel.ts]:
```suggestion
            if (!await isThemeCompiled(tmpPage)) {
```

File: src/fixtures/DefaultSalesChannel.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ import {
     getSnippetSetId,
     getThemeId,
 } from '../services/ShopwareDataHelpers';
-import { isSaaSInstance } from '../services/ShopInfo';
+import { isSaaSInstance, themeIsCompiled } from '../services/ShopInfo';
```
Comment (pweyck) [inline src/fixtures/DefaultSalesChannel.ts]:
```suggestion
import { isSaaSInstance, isThemeCompiled } from '../services/ShopInfo';
```

**Review by Phil23 — APPROVED** [review]

**Pull request merged**